### PR TITLE
added documentation about the FHIR Package source

### PR DESCRIPTION
**Pull request by mmsmits**

### CODE REVIEW

File: fhir-package-source/package-source.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+.. _packge_source:
+
+FHIR Package Source
+---------------------------
+
+As of SDK version 4.0 you can use a FHIR Package source to resolve FHIR artifacts like ``StructureDefinitions``, ``ValueSets``, and ``CodeSystems`` from one ore multiple FHIR Package.
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:6]: "one or more FHIR Package" -> "one or more FHIR NPM  packages"   

I think the term "package" is quite overloaded, so using the qualifier NPM at least once will help to disambiguate.

File: fhir-package-source/package-source.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+This can be from the FHIR core packages, but it can be from any FHIR Package you like. These for FHIR artifacts are needed for example for validating resources. Or to generate snapshots of profiles.
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:7]: "These _for_ FHIR"?

File: fhir-package-source/package-source.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Initiate a FHIR Package Source 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+There are multiple ways to create a ``FhirPackageSource`` instance, that include the packages you want to use.
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:11]: "...that include the packages" = "to include the packages" ?

File: fhir-package-source/package-source.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+However, if you don't have the packages you want to resolve your artifacts from yet, you can specify a FHIR package server, and the names and versions (using an ``@``) of the packages to install the packages locally and create a ``FhirPackageSource`` immediately.
+
+.. code:: csharp
+	
+	var package1 = "hl7.fhir.r3.core.xml@3.0.2";
+	var package2 = "hl7.fhir.r3.expansions@3.0.2";
+	FhirPackageSource resolver = new("https://packages.simplifier.net", new string[] { package1, package 2});
+
+.. note:: You may need an active internet connection to use this. This function will first check if the specified packages are already installed locally before searching for the packages online.
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:28]: Maybe we should talk about the location where the packages are cached in this case?

File: fhir-package-source/package-source.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	
+
+How to resolve artifacts from packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The most common use case for the FHIR Package source is to resolve conformance resources like ``StructureDefinitions``, ``ValueSets``, and ``CodeSystems``  from specific FHIR packages.
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:33]: I'd add that it implements `IResourceResolver`

File: fhir-package-source/package-source.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+All conformance resources can be resolved by specifying their canonical url.
+
+.. code:: csharp
+
+	FhirPackageSource resolver = new(new string[] { package1, package 2});
+	var patientProfile = await resolver.ResolveByCanonicalUriAsync("http://hl7.org/fhir/StructureDefinition/Patient") as StructureDefinition;
+	var genderValueSet = await resolver.ResolveByCanonicalUriAsync("http://hl7.org/fhir/ValueSet/administrative-gender") as ValueSet;
+
+You can also find any resource by their type and id, or any file by their name or relative filepath.
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:43]: And here, you are talking about `IArtifactResolver`, maybe good to add.

File: fhir-package-source/package-source.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+	ConceptMap cms = resolver.FindConceptMaps(sourceUri: "http://hl7.org/fhir/ValueSet/data-absent-reason", targetUri: "http://hl7.org/fhir/ValueSet/v3-NullFlavor");
+
+	//Find NamingSystem by uniqueId
+	NamingSystem ns = resolver.FindNamingSystem("http://snomed.info/sct");
+
+
+Different FHIR Versions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is also a ``class`` in the SDK called ``CommonFhirPackageSource``. This ``class`` is located in the FHIR version agnostic part of the SDK, and this is actually where all the magic happens.
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:77]: You don't have to `class` the word class, I have not done so in the rest of the documentation.

File: fhir-package-source/package-source.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+``FhirPackageSource`` is just a small layer on top of ``CommonFhirPackageSource``, and provides it with version information using a ``ModelInspector`` depending on which FHIR version of the SDK you are using.
+In practice this means that you can't combine packages of different FHIR versions in a single ``FhirPackageSource``, because the operations will need to resolve to different FHIR models.
+
```
Comment (ewoutkramer) [inline fhir-package-source/package-source.rst:80]: As a reader I am wondering: why do you mention this?  If this is a problem, how do I solve this?  Does CommonFhirPackageSource vs FhirPackageSource have anything to do with that?  If so, what?  Maybe just add an example to show how `new FhirPackageSource()` is equivalent to `new CommonFhirPackageSource(typeof(Patient).Assembly) or something like that.   And explain why you mention the fact that you cannot combine.

**Review by ewoutkramer — CHANGES_REQUESTED** [review]

File: fhir-package-source/package-source.rst [hunk 9]
```diff
@@ -25,14 +25,14 @@ However, if you don't have the packages you want to resolve your artifacts from
 	var package2 = "hl7.fhir.r3.expansions@3.0.2";
 	FhirPackageSource resolver = new("https://packages.simplifier.net", new string[] { package1, package 2});
 
-.. note:: You may need an active internet connection to use this. This function will first check if the specified packages are already installed locally before searching for the packages online.
+.. note:: You may need an active internet connection to use this. This function will first check if the specified packages are already installed locally before searching for the packages online. FHIR NPM packages are installed by default in ``C:\Users\xxxx\.fhir\packages\`` (Windows) or ``/Users/{user}/.fhir/packages/`` (MacOS)
```
Comment (vadi2) [inline fhir-package-source/package-source.rst:28]:
```suggestion
.. note:: You may need an active internet connection to use this. This function will first check if the specified packages are already installed locally before searching for the packages online. FHIR NPM packages are installed by default in ``C:\Users\{user}\.fhir\packages\`` (Windows), ``/Users/{user}/.fhir/packages/`` (macOS), or ``~/.fhir/packages/`` (Linux).
```

File: fhir-package-source/package-source.rst [hunk 10]
```diff
@@ -74,9 +74,27 @@ Next to that, there are some specific functions for certain resource types.
...
+        }
+
+
+        ///<inheritdoc/>
+        public async Task<Resource?> ResolveByCanonicalUriAsync(string uri)
+        {
+            return await _resolver.ResolveByCanonicalUriAsync(uri).ConfigureAwait(false);
+        }
+
+We see that a ``CommonFhirPackageSource`` including the ModelInspector of the currect FHIR version is created in the constructor, and that all functions in ``FhirPackageSource`` actually just call it's ``CommonFhirPackageSource`` equivalent right away.
```
Comment (vadi2) [inline fhir-package-source/package-source.rst:96]:
```suggestion
We see that a ``CommonFhirPackageSource``, including the ModelInspector of the currect FHIR version is created in the constructor, and that all functions in ``FhirPackageSource`` actually just call their ``CommonFhirPackageSource`` equivalent right away.
```

**Review by vadi2 — COMMENTED** [review]
Just a few editorial improvements

**Comment by mmsmits** [comment]
> Just a few editorial improvements

Thanks @vadi2

**Review by ewoutkramer — APPROVED** [review]

**Pull request merged**